src/main/java/org/candlepin/subscriptions/resource/SubscriptionTableController.java

```diff
@@ -197,6 +197,11 @@ public SkuCapacityReport capacityReportBySku( // NOSONAR
             var value = entry.getValue();
             addTotalCapacity(subscription, measurementKey, value, inventory);
           });
+      var billingProvider =
+          Optional.ofNullable(subscription.getBillingProvider())
+              .orElse(BillingProvider.EMPTY)
+              .asOpenApiEnum();
+      inventory.setBillingProvider(billingProvider);
```

Comment:
From my point of view, the real issue is that we're initializing the capacity with no billing provider at first place here: https://github.com/RedHatInsights/rhsm-subscriptions/blob/bbc421401ac277431dd5f20660586a41b30958f9/src/main/java/org/candlepin/subscriptions/resource/SubscriptionTableController.java#L267. 

I think a possible better solution would be to initialize the capacity directly using the subscription (instead of the sky and the billing provider). 

Just to try it out and see what the CI says about it, I've updated this pull request with this change. Let me know what you think about it.

Reply:
This is the commit with the changes https://github.com/RedHatInsights/rhsm-subscriptions/pull/2815/commits/57357db9599a364c2ec9ef8efaa2f057ac3395f9. 

Please, feel free to drop it if you prefer the changes as it was for some reason.